### ISSUE TITLE
linker error: undefined reference to re2::RE2::RE2

### ISSUE THREAD

**Issue by t0b3** [PDAL/PDAL#3944]
**Describe the bug**
compiling PDAL v2.5.0 on gentoo I run into a linker error:
```
PointViewTest.cpp:(.text+0x2c2d): undefined reference to `re2::RE2::RE2(re2::StringPiece const&)'
```
<details>
```
$ ninja -v -l0
[...]
/usr/bin/x86_64-pc-linux-gnu-g++ -march=native -O2 -pipe -Wl,-O1 -Wl,--as-needed test/unit/CMakeFiles/pdal_test_support.dir/Support.cpp.o test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o -o bin/pdal_point_view_test  -Wl,-rpath,/var/tmp/portage/sci-libs/pdal-2.5.0/work/PDAL-2.5.0-src_build/lib64:/var/tmp/portage/sci-libs/pdal-2.5.0/work/PDAL-2.5.0-src_build/lib  lib64/libpdal_base.so.15.0.0  lib64/libpdal_util.so.15.0.0  lib/libgtest.so.1.10.0
FAILED: bin/pdal_point_view_test 
: && /usr/bin/x86_64-pc-linux-gnu-g++ -march=native -O2 -pipe -Wl,-O1 -Wl,--as-needed test/unit/CMakeFiles/pdal_test_support.dir/Support.cpp.o test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o -o bin/pdal_point_view_test  -Wl,-rpath,/var/tmp/portage/sci-libs/pdal-2.5.0/work/PDAL-2.5.0-src_build/lib64:/var/tmp/portage/sci-libs/pdal-2.5.0/work/PDAL-2.5.0-src_build/lib  lib64/libpdal_base.so.15.0.0  lib64/libpdal_util.so.15.0.0  lib/libgtest.so.1.10.0 && :
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o: in function `PointViewDeathTest_out_of_bounds_Test::TestBody()':
PointViewTest.cpp:(.text+0x2c2d): undefined reference to `re2::RE2::RE2(re2::StringPiece const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o: in function `std::_Sp_counted_ptr<testing::internal::RE const*, (__gnu_cxx::_Lock_policy)2>::_M_dispose()':
PointViewTest.cpp:(.text._ZNSt15_Sp_counted_ptrIPKN7testing8internal2REELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt15_Sp_counted_ptrIPKN7testing8internal2REELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0xe): undefined reference to `re2::RE2::~RE2()'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o: in function `decltype (((testing::internal::MatcherBase<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::ValuePolicy<testing::MatcherInterface<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> const*, true>::Get({parm#1})).MatchAndExplain)({parm#2}, {parm#3})) testing::internal::MatcherBase<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::MatchAndExplainImpl<testing::internal::MatcherBase<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::ValuePolicy<testing::MatcherInterface<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> const*, true> >(testing::internal::MatcherBase<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&> const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, testing::MatchResultListener*)':
PointViewTest.cpp:(.text._ZN7testing8internal11MatcherBaseIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE19MatchAndExplainImplINSA_11ValuePolicyIPKNS_16MatcherInterfaceIS9_EELb1EEEEEDTcldtclsrT_3Getfp_E15MatchAndExplainfp0_fp1_EERKSA_S9_PNS_19MatchResultListenerE[_ZN7testing8internal11MatcherBaseIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE19MatchAndExplainImplINSA_11ValuePolicyIPKNS_16MatcherInterfaceIS9_EELb1EEEEEDTcldtclsrT_3Getfp_E15MatchAndExplainfp0_fp1_EERKSA_S9_PNS_19MatchResultListenerE]+0x55): undefined reference to `re2::RE2::PartialMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: PointViewTest.cpp:(.text._ZN7testing8internal11MatcherBaseIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE19MatchAndExplainImplINSA_11ValuePolicyIPKNS_16MatcherInterfaceIS9_EELb1EEEEEDTcldtclsrT_3Getfp_E15MatchAndExplainfp0_fp1_EERKSA_S9_PNS_19MatchResultListenerE[_ZN7testing8internal11MatcherBaseIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE19MatchAndExplainImplINSA_11ValuePolicyIPKNS_16MatcherInterfaceIS9_EELb1EEEEEDTcldtclsrT_3Getfp_E15MatchAndExplainfp0_fp1_EERKSA_S9_PNS_19MatchResultListenerE]+0x71): undefined reference to `re2::RE2::FullMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o: in function `testing::PolymorphicMatcher<testing::internal::MatchesRegexMatcher>::MonomorphicImpl<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::MatchAndExplain(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, testing::MatchResultListener*) const':
PointViewTest.cpp:(.text._ZNK7testing18PolymorphicMatcherINS_8internal19MatchesRegexMatcherEE15MonomorphicImplIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE15MatchAndExplainESC_PNS_19MatchResultListenerE[_ZNK7testing18PolymorphicMatcherINS_8internal19MatchesRegexMatcherEE15MonomorphicImplIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE15MatchAndExplainESC_PNS_19MatchResultListenerE]+0x3a): undefined reference to `re2::RE2::FullMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: PointViewTest.cpp:(.text._ZNK7testing18PolymorphicMatcherINS_8internal19MatchesRegexMatcherEE15MonomorphicImplIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE15MatchAndExplainESC_PNS_19MatchResultListenerE[_ZNK7testing18PolymorphicMatcherINS_8internal19MatchesRegexMatcherEE15MonomorphicImplIRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE15MatchAndExplainESC_PNS_19MatchResultListenerE]+0x61): undefined reference to `re2::RE2::PartialMatchN(re2::StringPiece const&, re2::RE2 const&, re2::RE2::Arg const* const*, int)'
/usr/lib/gcc/x86_64-pc-linux-gnu/11/../../../../x86_64-pc-linux-gnu/bin/ld: test/unit/CMakeFiles/pdal_point_view_test.dir/PointViewTest.cpp.o: in function `PointViewDeathTest_out_of_bounds_Test::TestBody() [clone .cold]':
PointViewTest.cpp:(.text.unlikely+0xa3e): undefined reference to `re2::RE2::~RE2()'
```
</details>

**Reason**
libre2.so is not being linked in this step
https://github.com/PDAL/PDAL/blob/366aa8442db3b5036fbe95ab019231a32c7c3777/test/unit/CMakeLists.txt#L76-L81

**Expected behavior**
compile and link w/o any errors

**System/installation information:**
gentoo with re2-2022.12.01

**Comment by abellgithub** [comment]
I don't understand why this would be building with debug but also optimization. Seems like a cmake configuration issue. I think you'd have to provide your command and the output of cmake.

**Comment by t0b3** [comment]
log file [pdal-2.5.0.log](https://github.com/PDAL/PDAL/files/10454236/pdal-2.5.0.log)
is the result with [pdal-2.5.0.ebuild.txt](https://github.com/PDAL/PDAL/files/10454246/pdal-2.5.0.ebuild.txt)
while for pdal-2.4.3 this ebuild worked like a charm is doesn't no longer. would I need to add any new cmake options? your input would be very valuable for me as a gentoo maintainer :smile:

**Comment by t0b3** [comment]
Closed by #3948 :rocket:
